lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
 
-    component = getMountedComponent({ variant: 'primary' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'primary');
+    rerender(<NxDropdown {...minimalProps} variant="primary" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--primary');
 
-    component = getMountedComponent({ variant: 'secondary' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'secondary');
+    rerender(<NxDropdown {...minimalProps} variant="secondary" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--secondary');
 
-    component = getMountedComponent({ variant: 'error' });
-    expect(component.find(NxButton)).toHaveProp('variant', 'error');
+    rerender(<NxDropdown {...minimalProps} variant="error" />);
+    expect(screen.getByRole('button')).toHaveClass('nx-btn--error');
   });
 
   it('correctly renders the menu based on isOpen prop', function() {
-    let component = getMountedComponent({ isOpen: true });
-    expect(component.find(NxDropdownMenu)).toExist();
+    const { container, rerender } = quickRender({ ...minimalProps, isOpen: true });
+    const menu = container.querySelector('.nx-dropdown-menu');
 
-    component = getMountedComponent({ isOpen: false });
-    expect(component.find(NxDropdownMenu)).not.toExist();
+    expect(menu).toBeInTheDocument();
+    rerender(<NxDropdown {...minimalProps} isOpen={false} />);
+    expect(menu).not.toBeInTheDocument();
   });
 
-  it('correctly assigns supplied classes', function() {
-    const component = getShallowComponent({ className: 'class1 class2' });
-    expect(component).toHaveClassName('nx-dropdown class1 class2');
+  it('sets the provided className', function() {
+    const el = renderEl()!;
+    const customEl = renderEl({ className: 'foo' })!;
+
+    expect(customEl).toHaveClass('foo');
+
+    for (const cls of Array.from(el.classList)) {
+      expect(customEl).toHaveClass(cls);
+    }
   });
 
-  it('disables the button (and the toggle fn) when the disabled prop is supplied', function() {
-    const toggleFn = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse: toggleFn, disabled: true }),
-        button = component.find(NxButton);
+  it('disables the button (and the toggle fn) when the disabled prop is supplied', async function() {
+    const user = userEvent.setup();
+    const toggleFn = jest.fn();
+    renderEl({ onToggleCollapse: toggleFn, disabled: true });
+    const button = screen.getByRole('button');
+    expect(button).toHaveClass('disabled');
+
+    await user.click(button);
 
-    expect(button).toHaveClassName('disabled');
-    button.simulate('click');
     expect(toggleFn).not.toHaveBeenCalled();
   });
 
-  it('uses VDOM as label if supplied', function() {
-    const label = <NxFontAwesomeIcon icon={faTrash} />,
-        component = getMountedComponent({ label });
-    const button = component.find(NxButton);
+  it('renders react element as label if supplied', function() {
```

Comment:
We should probably also test when `label` is a plain string

Reply:
> We should probably also test when `label` is a plain string

Added that test here: b02bf2ba530f04db03a59c4b6c75c64294cbb19d